sql/parse/dateparse/parsers.go

```diff
@@ -5,12 +5,20 @@ import (
 	"time"
 )
 
+type ParseType int
```

Comment:
Similar comment for this type... if there's an existing type we could reuse, that would be ideal. 

If we can't for some reason (import cycle?), then lets talk some more and see if we can avoid having `ParseType` and `OutType`, see if we can avoid exporting, and see if we can come up with more descriptive type names.